lib/cisco_node_utils/dns_domain.rb

```diff
@@ -59,8 +59,24 @@ def create
         config_set('dnsclient', 'domain_list',
                    state: '', name: @name)
       else
-        config_set('dnsclient', 'domain_list_vrf',
-                   state: '', name: @name, vrf: @vrf)
+        # On some platforms attempts to create a new domain-list results
+        # in the error. 'ERROR: Deletion of VRF test in progresswait
+        # for it to complete'.  We handle this by trying up to 500 times
+        # before giving up.
+        tries = 500
```

Comment:
Was hoping you would bring that up :).  Yes, I am trying as fast as possible and wanted your feedback.  It takes around 120 attempts which is just around 1-3 seconds.  Very fast.  I hesitated to add a sleep because I wanted it to bail out as quickly as possible but I am fine with a 1 second sleep.  Thoughts?